provider/rate_limiting/resource_rate_limiting_rule_alert.go

```diff
@@ -0,0 +1,953 @@
+		if blockingConfig,ok := firstThresholdActions["block"].(map[string]interface{}); ok {			
+			if blockingSeverity,ok := blockingConfig["eventSeverity"].(string); ok {
+				if blockingSeverity!=""{
+					d.Set("alert_severity",blockingSeverity)
+				}
+			}
+		}
+		thresholdConfigs := firstThresholdActionConfigs["thresholdConfigs"].([]interface{})
+		finalThresholdConfigs := []map[string]interface{}{}
+		for _,thresholdConfig := range thresholdConfigs {
+			var count_allowed float64
+			duration := ""
+			dynamic_mean_calculation_duration:=""
+			thresholdConfigData := thresholdConfig.(map[string]interface{})
+			if rollingWindowThresholdConfig,ok := thresholdConfigData["rollingWindowThresholdConfig"].(map[string]interface{}); ok {
+				if rollingWindowCountAllowed,ok := rollingWindowThresholdConfig["countAllowed"].(float64); ok{
+					count_allowed=rollingWindowCountAllowed
+				}
+				if rollingWindowDuration,ok := rollingWindowThresholdConfig["duration"].(string); ok{
+					duration=rollingWindowDuration
+				}
+
+			}
+			if dynamicThresholdConfig,ok := thresholdConfigData["dynamicThresholdConfig"].(map[string]interface{}); ok {
+				if dynamicCountAllowed,ok := dynamicThresholdConfig["percentageExceedingMeanAllowed"].(float64); ok{
+					count_allowed=dynamicCountAllowed
+				}
+				if dynamicDuration,ok := dynamicThresholdConfig["duration"].(string); ok{
+					duration=dynamicDuration
+				}
+				if dynamicMeanCalculationDuration,ok := dynamicThresholdConfig["meanCalculationDuration"].(string); ok{
+					dynamic_mean_calculation_duration=dynamicMeanCalculationDuration
+				}
+			}
+			thresholdConfigDataMap := map[string]interface{}{
+				"api_aggregate_type" : thresholdConfigData["apiAggregateType"].(string),
+				"rolling_window_count_allowed" : count_allowed,
+				"rolling_window_duration" : duration,
+				"threshold_config_type" : thresholdConfigData["thresholdConfigType"].(string),
+				"dynamic_mean_calculation_duration" : dynamic_mean_calculation_duration,
+			}
+			finalThresholdConfigs=append(finalThresholdConfigs,thresholdConfigDataMap)
+		}
+		d.Set("threshold_configs",finalThresholdConfigs)
+	}
+	conditionsArray := ruleDetails["conditions"].([]interface{})
+	finalReqResConditionsState := []map[string]interface{}{}
+	finalAttributeBasedConditionState := []map[string]interface{}{}
+
+	labelIdScopeFlag, endPointIdScopeFlag, ipReputationScopeFlag, ipLocationTypeScopeFlag, ipAbuseVelFlag,ipAddressFlag, emailDomainFlag, userAgentFlag, regionFlag, ipOrgFlag, ipAsnFlag, ipConnTypeFlag,reqScannerFlag, userIdFlag := true, true, true, true, true, true, true, true, true, true, true, true, true, true
+	for _,condition := range conditionsArray {
+		leafCondition := condition.(map[string]interface{})["leafCondition"].(map[string]interface{})
+		conditionType := leafCondition["conditionType"].(string)
+		finalConditionState := []map[string]interface{}{}
+		if conditionType == "IP_REPUTATION"{
```

Comment:
Since there are so many if elses, better to implement using switch case. Also this file needs to be reformatted.